examples/standalone/runfile/dynamics.py

```diff
@@ -37,84 +38,112 @@
         action="store",
         help="path to the namelist",
     )
-    args = parser.parse_args()
-    backend = args.backend
-    data_dir = args.data_dir
-    time_step = args.time_step
-
-    # # MPI stuff
-    comm = mpi4py.MPI.COMM_WORLD
-    rank = comm.Get_rank()
-
-    # fv3core specific setup
-    fv3core.set_backend(backend)
-    fv3core.set_rebuild(False)
-
-    # namelist setup
-    spec.set_namelist(data_dir + "/input.nml")
-
-    nml2 = yaml.safe_load(
-        open(
-            data_dir + "/input.yml",
-            "r",
-        )
-    )["namelist"]
-    experiment_name = yaml.safe_load(
-        open(
-            data_dir + "/input.yml",
-            "r",
-        )
-    )["experiment_name"]
 
-    # set up of helper structures
-    serializer = serialbox.Serializer(
-        serialbox.OpenModeKind.Read,
-        data_dir,
-        "Generator_rank" + str(rank),
-    )
-    cube_comm = util.CubedSphereCommunicator(
-        comm, util.CubedSpherePartitioner.from_namelist(nml2)
-    )
+    return parser.parse_args()
+
+
+def set_experiment_info(experiment_name, time_step, backend):
+    experiment = {}
+    sha = git.Repo(
+        pathlib.Path(__file__).parent.absolute(), search_parent_directories=True
+    ).head.object.hexsha
+    now = datetime.now()
+    dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
+    experiment["setup"] = {}
+    experiment["setup"]["timestamp"] = dt_string
+    experiment["setup"]["dataset"] = experiment_name
+    experiment["setup"]["timesteps"] = time_step
+    experiment["setup"]["hash"] = sha
+    experiment["setup"]["version"] = "python/" + backend
+    experiment["times"] = {}
+    return experiment
+
+
+def gather_timing_statistics(timer, experiment, comm, root=0):
+    is_root = comm.Get_rank() == root
+    recvbuf = np.array(0.0)
+    for name, value in timer.times.items():
+        if is_root:
+            print(name)
+            experiment["times"][name] = {}
+        for label, op in [
+            ("minimum", MPI.MIN),
+            ("maximum", MPI.MAX),
+            ("mean", MPI.SUM),
+        ]:
+            comm.Reduce(np.array(value), recvbuf, op=op)
+            if is_root:
+                if label == "mean":
+                    recvbuf /= comm.Get_size()
+                print(f"    {label}: {recvbuf}")
+                experiment["times"][name][label] = float(recvbuf)
+
+
+def write_global_timings(experiment, comm, root=0):
```

Comment:
I would suggest refactoring this so one of its arguments is either a filename or (better) an open file object. Might mean writing another `get_filename` helper function.